### PR TITLE
Task #5 Class Frame and Game

### PR DESCRIPTION
**Pull request by Angan7a**
Added class Frame and Game.
Some method in Game need implementation

### CODE REVIEW

File: tests/FrameTests.cpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include <gtest/gtest.h>
+#include "../inc/Frame.hpp"
+
+struct FrameTests : public ::testing::Test
+{
+    Frame frame;
+};
+
+TEST_F(FrameTests, checkIfIncicalValueIsZero)
```
Comment (Tomek52) [inline tests/FrameTests.cpp]: Angan7a, please add
`//given //when //then` 
to unit tests

File: inc/Frame.hpp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+class Frame
+{
+    int value_;
+    bool isStrike;
```
Comment (Tomek52) [inline inc/Frame.hpp]: Angan7a, I think this bool variables can be const, cause information strike and spare will be known when object will be created

File: inc/Frame.hpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    bool isSpare;
+public:
+    Frame();
+    int getValue();
```
Comment (Tomek52) [inline inc/Frame.hpp]: Angan7a, this function can be const

File: src/Frame.cpp [hunk 4]
```diff
@@ -4,6 +4,12 @@ Frame::Frame() : value_(0)
 {
 
 }
+
+void Frame::setValue(int value)
+{
+    value_ = value;
```
Comment (Tomek52) [inline src/Frame.cpp]: Angan7a, maybe better write value_ += value?

File: inc/Game.hpp [hunk 5]
```diff
@@ -1,6 +1,10 @@
 #pragma once
+#include <string>
 
 class Game
 {
+    std::string name_;
+public:
+    std::string getName();
```
Comment (Tomek52) [inline inc/Game.hpp]: Angan7a, this function can be const

**Comment by Angan7a** [comment]
Tomek I made changes. Please check.

**Comment by Tomek52** [comment]
@Angan7a good job :D

File: inc/Game.hpp [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+#include <string>
+#include "../inc/Frame.hpp"
+
+class Game
+{
+    std::string name_;
+    Frame frame[12];
+public:
+    std::string getName() const;
+    void setName(const std::string &);
```
Comment (LordLukin) [inline inc/Game.hpp]: Why not to set the game name in constructor only? Do we assume that the name will be changed? Or will we create a new object for a new game?

File: inc/Game.hpp [hunk 7]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include <string>
+#include "../inc/Frame.hpp"
```
Comment (LordLukin) [inline inc/Game.hpp]: you could add in CMakeLists.txt include_direcories(inc) and then you will be able to simply #include "Frame.hpp"

File: tests/FrameTests.cpp [hunk 8]
```diff
@@ -0,0 +1,47 @@
+#include <gtest/gtest.h>
+#include "../inc/Frame.hpp"
+
+struct FrameTests : public ::testing::Test
+{
+    //GIVEN
+    Frame frame;
+};
+
+TEST_F(FrameTests, check_if_incical_value_is_zero)
```
Comment (LordLukin) [inline tests/FrameTests.cpp]: typo: incical

File: inc/Game.hpp [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+class Game
+{
+    std::string name_;
+    Frame frame[12];
+public:
+    std::string getName() const;
+    void setName(const std::string &);
+    int score() const;
+    int checkValue(const std::string & value);
```
Comment (LordLukin) [inline inc/Game.hpp]: In TDD you should not write any code that is not used. So checkValue, checkIsStrikeOrSpare and setValueFrameAndPrevFrames are not needed. YAGNI.

**Pull request merged**